Client/Frontend/Browser/Tabs/TabTrayViewController.swift

```diff
@@ -2,12 +2,111 @@
+    var themeObserver: NSObjectProtocol?
+    var notificationCenter: Common.NotificationProtocol
+
+    var openInNewTab: ((URL, Bool) -> Void)?
+    var didSelectUrl: ((URL, Storage.VisitType) -> Void)?
+
+    private lazy var navigationToolbar: UIToolbar = .build { [self] toolbar in
+        toolbar.delegate = self
+        toolbar.setItems([UIBarButtonItem(customView: segmentedControl)], animated: false)
+        toolbar.isTranslucent = false
+    }
+
+    private lazy var segmentedControl: UISegmentedControl = {
+        return createSegmentedControl(action: #selector(segmentChanged),
+                                      a11yId: AccessibilityIdentifiers.TabTray.navBarSegmentedControl)
+    }()
+
+    lazy var countLabel: UILabel = {
+        let label = UILabel(frame: CGRect(width: 24, height: 24))
+        label.font = TabsButton.UX.titleFont
+        label.layer.cornerRadius = TabsButton.UX.cornerRadius
+        label.textAlignment = .center
+        // TODO: Connect to regular tabs count
+        label.text = "0"
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    var segmentControlItems: [Any] {
+        let iPhoneItems = [
+            LegacyTabTrayViewModel.Segment.tabs.image!.overlayWith(image: countLabel),
+            LegacyTabTrayViewModel.Segment.privateTabs.image!,
+            LegacyTabTrayViewModel.Segment.syncedTabs.image!]
+        return shouldUseiPadSetup() ? LegacyTabTrayViewModel.Segment.allCases.map { $0.label } : iPhoneItems
+    }
+
+    init(themeManager: ThemeManager = AppContainer.shared.resolve(),
+         and notificationCenter: NotificationProtocol = NotificationCenter.default) {
+        self.themeManager = themeManager
+        self.notificationCenter = notificationCenter
+
+        super.init(nibName: nil, bundle: nil)
+        self.applyTheme()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupView()
+    }
+
+    func applyTheme() {
+        view.backgroundColor = themeManager.currentTheme.colors.layer1
+        navigationToolbar.barTintColor = themeManager.currentTheme.colors.layer1
+    }
+
+    private func setupView() {
+        view.addSubview(navigationToolbar)
+        navigationToolbar.setItems([UIBarButtonItem(customView: segmentedControl)], animated: false)
+
+        NSLayoutConstraint.activate([
+            navigationToolbar.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            navigationToolbar.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            navigationToolbar.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ])
+    }
+
+    private func createSegmentedControl(
+        action: Selector,
+        a11yId: String
+    ) -> UISegmentedControl {
+        let segmentedControl = UISegmentedControl(items: segmentControlItems)
```

Comment:
So interestingly, I have a task on my end for the Component library to add a SegmentedControl component. Design sees [this](https://www.figma.com/file/YIbYarab5aYte7KbK0bcVw/iOS-Components?type=design&node-id=1507-8124&mode=design&t=ZnctcivAJ40xSPfd-0) as a reusable component. Do you think it's worth adding now in the library instead?

Reply:
I talked with Orla about a possible UI Component library elements but this case is straightforward and other than `applyTheme` there is nothing extra from UIKitSegmentControl so I'm not sure it's worthy 🤔